sdk/core/core-amqp/test/requestResponse.spec.ts

```diff
@@ -136,6 +136,60 @@ describe("RequestResponseLink", function() {
     assert.equal(responses[1].correlation_id, reqs[1].message_id);
   });
 
+  it("request without `message_id` should not match a response with `undefined` correlationId", async function() {
```

Comment:
I don't understand the point of this test case...

I would support adding a test case to ensure that when sendRequest gets a request without a message id, we should set one. After this point, other tests in this file would test that the right response is matched with the right request.

Reply:
> I don't understand the point of this test case...

The test would've failed previously (if executed without the changes to core-amqp in this PR).
I feel this is needed to avoid regressions.

> I would support adding a test case to ensure that when sendRequest gets a request without a message id, we should set one.

Thanks. Added an additional check in the same test to make sure of this.

Reply:
Wouldnt the regression be avoided by the test that checks that the request always has a message id?

Reply:
Hmmm.. I don't have strong reasons to keep my test now, can remove, and just check for message_id.

Reply:
Updated - 71debde
Good now?